src/AggregateFunctions/AggregateFunctionGroupArraySorted.h

```diff
@@ -0,0 +1,390 @@
+#pragma once
```

Comment:
There is a lot of overlap (copypasta) with AggregateFunctions/AggregateFunctionGroupArray.h/cpp. Is it possible to integrate below code into the existing files, i.e. basically add a new bool to the `Trait` class and then handle sorting under new `constexpr if`s?

Reply:
Sorry to unresolve the comment, but >400 LOC boilerplate code are added where it is not strictly necessary. Could you please check if the new functionality can be covered by extending the existing AggregateFunctionGroupArray.h/cpp?

Reply:
I am ok about the copy paste, cause the code will be simpler, and some unneeded things like sampling are removed. And in-memory state could be a bit smaller.

But the code need to be cleaned.